src/ray/raylet/node_manager.cc

```diff
@@ -2093,17 +2079,7 @@ void NodeManager::ForwardTask(const Task &task, const ClientID &node_id,
       fbb.GetBufferPointer(),
       [this, on_error, task_id, node_id, spec](ray::Status status) {
         if (status.ok()) {
-          // If we were able to forward the task, remove the forwarded task from the
-          // lineage cache since the receiving node is now responsible for writing
-          // the task to the GCS.
-          if (!lineage_cache_.RemoveWaitingTask(task_id)) {
-            RAY_LOG(WARNING) << "Task " << task_id << " already removed from the lineage"
-                             << " cache. This is most likely due to reconstruction.";
-          } else {
-            // Mark as forwarded so that the task and its lineage is not
-            // re-forwarded in the future to the receiving node.
-            lineage_cache_.MarkTaskAsForwarded(task_id, node_id);
```

Comment:
`MarkTaskAsForwarded` is an optimization to avoid sending duplicate lineage, right? If we flush tasks much more aggressively, then it may not be necessary.

I'm mildly concerned about this case because it's possible to end up in this case where `status.ok()` returns `true` but the forwarding actually fails. It may not be an issue, because in that case one of the two nodes probably failed, but I'm not sure.